LTP:Fixed issue setfsuid syscall test setfsuid04

vamsikrishna935 opened the pull request:
Testcase met success criteria but due to lack of child process, it is
not reporting test status, so replaced exit with tst_resm to report
the status and removed wait for child in parent.

Commit:
LTP:Fixed issue setfsuid syscall test setfsuid04

Testcase met success criteria but due to lack of child process, it is
not reporting test status, so replaced exit with tst_resm to report
the status and removed wait for child in parent.

letmaik commented:
Can you explain why exit() doesn't work?

vamsikrishna935 commented:
> 
> 
> Can you explain why exit() doesn't work?

exit() will return the status to parent process here we don't have 2 process, so in the single process environment it is exiting improperly, so modified the tests to work for single process environment.

letmaik commented:
I see, but what does "exiting improperly" mean?

vamsikrishna935 commented:
> 
> 
> I see, but what does "exiting improperly" mean?

The status reporting from exit is not happening.

letmaik commented:
What does that mean? There's a printf before the exit, and the exit with non-zero code should mark the test as failed. What am I missing?

vamsikrishna935 commented:
> 
> 
> What does that mean? There's a printf before the exit, and the exit with non-zero code should mark the test as failed. What am I missing?

Printf statement will print the message, usually exit() method sends the status to parent process, but we have only one process so the status from exit will not be reported. Because of this test result is showing as skipped. So as we cannot report the status from printf and in order to represent test status , i changed it to tst_resm.

vamsikrishna935 commented:
> 
> 
> > What does that mean? There's a printf before the exit, and the exit with non-zero code should mark the test as failed. What am I missing?
> 
> Printf statement will print the message, usually exit() method sends the status to parent process, but we have only one process so the status from exit will not be reported. Because of this test result is showing as skipped. So as we cannot report the status from printf and in order to represent test status , i changed it to tst_resm.

Result of this testcase from CI Run : 'kernel-syscalls-setfsuid-setfsuid04' **skipped**. Failure Count = 0, Pass Count = 0.

Our reporting mechanism uses Pass Count and Failure count to report the test status as passed / failed. Here both counts are zero , so it will be marked as skipped. So i have modified it to get pass/fail counts.

letmaik commented:
So do I understand correctly that you do not look at the exit code at all but instead only scan for keywords in the output?

hukoyu commented:
@vamsikrishna935  if you are talking about run_ltp_test.sh (https://github.com/lsds/sgx-lkl/blob/oe_port/tests/ltp/run_ltp_test.sh) there is no more skipped tests. There are failures or passes.

vamsikrishna935 commented:
> 
> 
> So do I understand correctly that you do not look at the exit code at all but instead only scan for keywords in the output?

The reporting framework is designed to look at the output and it is generalized for all the tests.

> 
> 
> @vamsikrishna935 if you are talking about run_ltp_test.sh (https://github.com/lsds/sgx-lkl/blob/oe_port/tests/ltp/run_ltp_test.sh) there is no more skipped tests. There are failures or passes.

But we need pass count/ fail count to report the status right ?

letmaik commented:
> > So do I understand correctly that you do not look at the exit code at all but instead only scan for keywords in the output?
> 
> The reporting framework is designed to look at the output and it is generalized for all the tests.

If it was general then you wouldn't have to patch output strings in the test cases. Why can't we additionally look at the exit code as well to avoid patching strings?

hukoyu commented:
> > > So do I understand correctly that you do not look at the exit code at all but instead only scan for keywords in the output?
> > 
> > 
> > The reporting framework is designed to look at the output and it is generalized for all the tests.
> 
> If it was general then you wouldn't have to patch output strings in the test cases. Why can't we additionally look at the exit code as well to avoid patching strings?

@letmaik current LTP test runner just looks at pass failure keywords and doesn't look at the exit code. My guess is relying on exit code for these tests doesn't work in all cases like this one since there is child process/parent process cases. I strongly believe using exit code is out of scope of enabling these test cases. Let's stick with current architecture and logic and I will try and submit a PR on run_ltp_test.sh using exit code and will see if we will have test failures due to that,

hukoyu commented:
@letmaik with current test pass and fail logic I asked PASS/TEST_FAILED outputs in the tests from PR createors.

vamsikrishna935 commented:
> 
> 
> > > So do I understand correctly that you do not look at the exit code at all but instead only scan for keywords in the output?
> > 
> > 
> > The reporting framework is designed to look at the output and it is generalized for all the tests.
> 
> If it was general then you wouldn't have to patch output strings in the test cases. Why can't we additionally look at the exit code as well to avoid patching strings?


> 
> 
> > > So do I understand correctly that you do not look at the exit code at all but instead only scan for keywords in the output?
> > 
> > 
> > The reporting framework is designed to look at the output and it is generalized for all the tests.
> 
> If it was general then you wouldn't have to patch output strings in the test cases. Why can't we additionally look at the exit code as well to avoid patching strings?

@letmaik 
Let me brief you the changes i have made:
The current framework will not look exit() code for the status.
We are having PASSED/PASS, TEST_FAILED keywords to state the status.
Changes :
The testcase is implemented using parent and child processes. All the tests were happening in child block.
Usually exit() method will report the status to the parent process. But we have only one process, so the status reporting from exit will not work here. So removed the exit() code.
Also the test does not have PASSED/TEST_FAILED keywords, so i have added those keywords in respective places. Instead of printf , i have used tst_resm function because it is a function designed in framework for status report using (TFAIL/TPASS).
Also commented parent waiting for the child status.

Commit:
Removed PASSED/TEST_FAILED Keywords.

vamsikrishna935 commented:
@letmaik : I have removed PASSED/TEST_FAILED keywords, could you please review .
